module/assets/helper.js

```diff
@@ -0,0 +1,849 @@
+          );
+          if (checkRoleToUpdateProgram) {
+            roleDetails = {
+              roleId: roleId,
+              code: roleCode.code,
+              programs: roleCode.programs.filter((eachProgram) =>
+                eachProgram.equals(reqData.assetInformation.identifier)
+              ),
+            };
+          }
+        } else {
+          roleDetails = {
+            roleId: roleId,
+            code: roleCode.code,
+            programs: roleCode.programs,
+          };
+        }
+
+        if (
+          roleCode.code === constants.common.PROGRAM_DESIGNER &&
+          roleDetails
+        ) {
+          roleDetails.isAPlatformRole = true;
+          roleDetails.entities = [];
+        }
+        if (roleDetails) {
+          programRolesArray.push(roleDetails);
+        }
+      }
+    }
+
+    return programRolesArray;
+  }
+
+  /**
+   * Check whether from and to user has Identical Roles.
+   * @method
+   * @name checkRolesPresence
+   * @param {Array} fromUserData  -fromUser array.
+   * @param {Array} toUserArray   - toserU array.
+   * @returns {Boolean} returns true or false.
+   */
+
+  static checkRolesPresence(fromUserRoleArray, toUserRoleArray) {
+    let rolesToCheck = [
+      constants.common.PROGRAM_MANAGER,
+      constants.common.PROGRAM_DESIGNER,
+      constants.common.CONTENT_CREATOR,
+    ];
+
+    let hasRolesInFromArray = rolesToCheck.some((role) =>
+      fromUserRoleArray.includes(role)
+    );
+    let hasRolesInToArray = rolesToCheck.some((role) =>
+      toUserRoleArray.includes(role)
+    );
+
+    return hasRolesInFromArray && hasRolesInToArray;
+  }
+  /**
+ * Query to get Platform Roles for touser when user Not in UserExtension 
+ * @method
+ * @name fetchFromUserDataPlatformRoles
+ * @param {Array} fromUserData -   From user data
+ * @returns {Array} returns Array of Platform Roles to push for touser
+ * @returns
+ * 
+     * 
+    * [
+          {
+            roleId: 60c74aa8b81797534e9a3f11,
+            code: 'PROGRAM_MANAGER',
+            programs: [
+              607d320de9cce45e22ce90c0,
+            ]
+          },
+          {
+            roleId: 60c83b247f8464532732cdc3,
+            code: 'PROGRAM_DESIGNER',
+            programs: [
+                601429016a1ef53356b1d714,
+            ],
+            isAPlatformRole: true,
+            entities: []
+            }
+          ]
+    */
+  static fetchFromUserDataPlatformRoles(
+    fromUserData,
+    reqData,
+    checkAssetInformation
+  ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let filterCodes = fromUserData.platformRoles.map((role) => role.code);
+
+        let findProgramManagerQuery = {
+          code: { $in: filterCodes },
+        };
+
+        let findProgramManagerId = await userRolesHelper.roleDocuments(
+          findProgramManagerQuery
+        );
+
+        // returns ProgramRoles array
+        let programRolesArray = this.createProgramRolesArray(
```

Comment:
@praveenKDass - As discussed for all functions being called in this file, please add sample request and response values as comments.